modules/katello_sync_plan.py

```diff
@@ -84,154 +92,89 @@
-                needs_update = True
-        return needs_update, old
-
-    def sync_plan(self, name, organization, interval=None, enabled=None, sync_date=None, products=[]):
-        updated = False
-        organization = self.find_organization(organization)
-
-        enabled = enabled.lower() == "true"
-
-        sync_plan = self._entities.SyncPlan(self._server, name=name, organization=organization, interval=interval, enabled=enabled, sync_date=sync_date)
-        response = sync_plan.search({'name', 'organization'})
-
-        if len(response) == 1:
-            response[0].sync_date = parse_time(response[0].sync_date)
-            updated, sync_plan = self.update_fields(sync_plan, response[0], ['interval', 'enabled', 'sync_date'])
-            if updated and not self._module.check_mode:
-                sync_plan.update()
-        elif len(response) == 0:
-            if not self._module.check_mode:
-                sync_plan = sync_plan.create()
-            updated = True
-
-        desired_product_ids = list(map(lambda p: p.id, self.find_products(products, organization)))
-        current_product_ids = list(map(lambda p: p.id, sync_plan.product))
+RETURN = ''' # '''
+
+from ansible.module_utils.ansible_nailgun_cement import (
+    SyncPlan,
+    find_organization,
+    find_products,
+    find_sync_plan,
+    naildown_entity,
+    sanitize_entity_dict,
+)
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
 
-        if set(desired_product_ids) != set(current_product_ids):
-            if not self._module.check_mode:
-                product_ids_to_add = set(desired_product_ids) - set(current_product_ids)
-                if len(product_ids_to_add) > 0:
-                    sync_plan.add_products(data={'product_ids': list(product_ids_to_add)})
-                product_ids_to_remove = set(current_product_ids) - set(desired_product_ids)
-                if len(product_ids_to_remove) > 0:
-                    sync_plan.remove_products(data={'product_ids': list(product_ids_to_remove)})
-            updated = True
 
-        return updated
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'description': 'description',
+    'organization': 'organization',
+    'interval': 'interval',
+    'enabled': 'enabled',
+    'sync_date': 'sync_date',
+    'cron_expression': 'cron_expression',
+    'products': 'product',
+}
 
 
 def main():
-    module = ForemanAnsibleModule(
+    module = ForemanEntityAnsibleModule(
         argument_spec=dict(
             name=dict(required=True),
+            description=dict(),
             organization=dict(required=True),
-            interval=dict(required=True),
-            enabled=dict(required=True),
+            interval=dict(choices=['hourly', 'daily', 'weekly', 'custom cron'], required=True),
+            enabled=dict(type='bool', required=True),
             sync_date=dict(required=True),
```

Comment:
The docs says it has a default, but it doesn't ;)

Reply:
It also says, it is required, which doesn't work...